### PR TITLE
Support querying files by modification time and size

### PR DESCRIPTION
**Pull request by yetist**
Thanks for Leslie Zhai's patch, UI like this:

Please review and test it, thanks.

### CODE REVIEW

**Comment by xiangzhai** [comment]
@lukefromdc I am Leslie Zhai :)

> Some test searches seemed to hang

Perhaps it needs to optimize the algorithm, for example O(N), about search function especially for *huge* filesystem testcase.

Regards,
Leslie Zhai

**Comment by lukefromdc** [comment]
Caja can be set to list filesizes either by K/M/GiB or by K/M/GB in preferences, so the labels ideally would be following user preferences but that would require reading the gsettings value and then selecting which label text to use. 

File size comes from 
`gint64 file_size = g_file_info_get_size (info);`
I do not know which units that returns file size in.

File: src/caja-query-editor.c [hunk 1]
```diff
@@ -999,6 +1025,231 @@ type_add_rows_from_query (CajaQueryEditor    *editor,
...
+    gtk_list_store_append(duration_store, &iter);
+    gtk_list_store_set(duration_store, &iter, 0, 86400, 1, _("1 Day"), -1);
+    gtk_list_store_append(duration_store, &iter);
+    gtk_list_store_set(duration_store, &iter, 0, 604800, 1, _("1 Week"), -1);
+    gtk_list_store_append(duration_store, &iter);
+    gtk_list_store_set(duration_store, &iter, 0, 2419200, 1, _("1 Month"), -1);
+    gtk_list_store_append(duration_store, &iter);
+    gtk_list_store_set(duration_store, &iter, 0, 14515200, 1, _("6 Months"), -1);
+    gtk_list_store_append(duration_store, &iter);
+    gtk_list_store_set(duration_store, &iter, 0, 29030400, 1, _("1 Year"), -1);
```
Comment (vkareh) [inline src/caja-query-editor.c]: These values seem wrong. Here's what I noticed:
🗸 3600: 1 hour
🗸 86400: 1 day, 24 hours
🗸 604800: 7 days, 168 hours
🗴 2419200: 28 days, arguably correct?
🗴 14515200: 168 days, definitely wrong (should be around 15778800, 31557600 / 2, or about 183 days)
🗴 29030400: 336 days, definitely wrong (should be around 31557600, 365.25 days)

File: src/caja-query-editor.c [hunk 2]
```diff
@@ -999,6 +1025,231 @@ type_add_rows_from_query (CajaQueryEditor    *editor,
...
+    gtk_list_store_append(size_store, &iter);
+    gtk_list_store_set(size_store, &iter, 0, 104857600, 1, _("100 MB"), -1);
+    gtk_list_store_append(size_store, &iter);
+    gtk_list_store_set(size_store, &iter, 0, 524288000, 1, _("500 MB"), -1);
+    gtk_list_store_append(size_store, &iter);
+    gtk_list_store_set(size_store, &iter, 0, 1073741824, 1, _("1 GB"), -1);
+    gtk_list_store_append(size_store, &iter);
+    gtk_list_store_set(size_store, &iter, 0, 2147483648, 1, _("2 GB"), -1);
+    gtk_list_store_append(size_store, &iter);
+    gtk_list_store_set(size_store, &iter, 0, 4294967296, 1, _("4 GB"), -1);
```
Comment (vkareh) [inline src/caja-query-editor.c]: Agree with the comments above. These units (and their value) should reflect the users preference

**Comment by lukefromdc** [comment]
The filesize labels now follow user preferences. One minor quibble is this: on toggling "show file sizes with IEC units" in File Management Preferences>Display the labels do not update until the search results window has been toggled away from "size" and back  or the search closed and an new search started (which returns the search subwindow to "location" . 

Since it would probably not be common for a user to toggle this preference in the middle of a search I'm not sure if its worth the extra work (and code at runtime) to connect to the "changed" signal for that gsettings preference to update the labels or not. Still this would work, and would catch the case of a user doing a filesize search and deciding at that point to reset the filesize units preference for Caja to use a particular size cutoff.

**Comment by raveit65** [comment]
I got build warnings.
```
caja-query.c: In function 'caja_query_to_xml':
caja-query.c:387:12: warning: unused variable 'size' [-Wunused-variable]
     gint64 size;
            ^~~~
caja-query.c:386:12: warning: unused variable 'timestamp' [-Wunused-variable]
     gint64 timestamp;
            ^~~~~~~~~
```
```
caja-query-editor.c: In function 'modtime_row_add_to_query':
caja-query-editor.c:1162:17: warning: 'datetime' may be used uninitialized in this function [-Wmaybe-uninitialized]
     timestamp = g_date_time_to_unix (datetime);
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```
Not sure if the following is related to PR
```
caja-search-engine-simple.c: In function 'visit_directory':
caja-search-engine-simple.c:350:17: warning: unused variable 'tag' [-Wunused-variable]
     const char *tag, *mime_type, *display_name;
```

**Comment by yetist** [comment]
Okey, I'll check it.

**Comment by yetist** [comment]
@raveit65 
New PR here: #1127

**Pull request merged**